### PR TITLE
Pass flag for helm version rather than use env var

### LINKED ISSUE #195: Can not find tiller during app install of nginx-ingress
## Expected Behavior
Install of ingress-nginx

## Current Behaviour
$ k3sup app install nginx-ingress
Using kubeconfig: /home/pi/.kube/config
Using helm3
Client: armv7l, Linux
2020/02/14 08:13:02 User dir established as: /home/pi/.k3sup/
"stable" has been added to your repositories
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "stable" chart repository
Update Complete.
armNode architecture: "arm"
Chart path:  /tmp/charts
VALUES values.yaml
Command: /home/pi/.k3sup/bin/helm3/helm [upgrade --install nginx-ingress stable/nginx-ingress --namespace default --values /tmp/charts/nginx-ingress/values.yaml --set defaultBackend.enabled=false --set controller.image.repository=quay.io/kubernetes-ingress-controller/nginx-ingress-controller-arm]
Error: could not find tiller
Error: exit code 1, stderr: Error: could not find tiller

## Steps to Reproduce (for bugs)
1. k3sup install nginx-ingress

## Your Environment

* What Kubernetes distribution are you using (for `k3sup app`)?

1.17.2

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:30:10Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/arm"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:22:30Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/arm"}
```

* Operating System and version (e.g. Linux, Windows, MacOS):
```
$ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description
We were relying on each app setting an env var to
specify helm3 version, but if that wasn't set then
we installed helm2, and didn't try and init.
This left the user unable to use helm3 on any other
app too.

It has been changed to pass down a flag, then use that
flag in 1 place to decide what version to install.
Users can still override the version by setting an
env var, which is documented in the README.

This has been tested by rm -rf ing the ~/.k3sup
directory and installing using no flag, --helm3=false
, --helm3=true and --helm3

I have also overridden the version using the ENV Var

Fixes #195

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
#195

## How Has This Been Tested?

Using the flag, not using the flag and setting the env var for override. 
I deleted ~/.k3sup/ between each run

All worked with correct versions used now.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pkg/helm/helm.go [hunk 1]
```diff
@@ -52,8 +51,14 @@ func GetHelmURL(arch, os, version string) string {
...
-func DownloadHelm(userPath, clientArch, clientOS, subdir string) error {
+func DownloadHelm(userPath, clientArch, clientOS, subdir string, helm3 bool) error {
+
 	useHelmVersion := helmVersion
+	if helm3 {
+		useHelmVersion = helm3Version
+	}
+
+	// This is an override users can set, its documented in the README
 	if val, ok := os.LookupEnv("HELM_VERSION"); ok && len(val) > 0 {
```
Comment (alexellis) [inline pkg/helm/helm.go]: I'm still seeing HELM_VERSION being used here? Won't it have the same issues as when we set it in each app?

**Comment by alexellis** [comment]
We should never initialise tiller, the helm2 code should only ever do a client-side initialisation.

Can you point out the code that initialises tiller? With Helm2, we only ever run `helm init --client-only`

File: pkg/helm/helm.go (hunk 1 above)
Comment (Waterdrips) [reply pkg/helm/helm.go]: This is to keep the override option thats documented in the README, the apps no longer set the "HELM_VERSION"

**Comment by Waterdrips** [comment]
> We should never initialise tiller, the helm2 code should only ever do a client-side initialisation.
> 
> Can you point out the code that initialises tiller? With Helm2, we only ever run `helm init --client-only`

Yeh, we only init client-only - I must have got confused when writing commit message - ill remove reference to tiller.

**Comment by alexellis** [comment]
Looks like changes were pushed 👍 

Are you able to confirm the manual testing scenarios?

**Comment by Waterdrips** [comment]
> Looks like changes were pushed +1
> 
> Are you able to confirm the manual testing scenarios?

New k3d cluster before every k3sup command
The nginx-ingress app was used for testing. (All apps now rely on shared code to set and download helm)

Scenarios:
1)
rm -rf ~/.k3sup and install with --helm3,
then install with --helm3 without deleting ~/.l3sup
then install with --helm3=false wtihout deleting ~/.k3sup

2)
rm -rf ~/.k3sup and install with --helm3=false
then install with --helm3 without deleting ~/.l3sup
then install with --helm3=false without deleting ~/.k3sup


The output was checked to see the correct version was installed (when it didn't exist, after deleting the ~/.k3sup dir), when the dir was there, and the specific helm version was already installed, we didnt try and reinstall, but the command succeed.

**Pull request merged**